### PR TITLE
feat: social news feed

### PR DESCRIPTION
**Pull request by sakul-budhathoki**
### Article creation http://deploy-preview-318--testitori.netlify.app/feed/new
- Can't add audio in the RichText for now, so they are are added at the bottom of the Rich Text
- Can't load respective players for each streaming plateform from an URL. Disable OpenGraph stuff for now.
- The videos, images and gifs can be added only by their respective buttons (Upload), and are added into the Rich Text.

### Mentions
- We can add a mention by writting "@blabla"
- Then, on a user profile, we can see the the posts that contain his user id or wallet address in in their metadata `mentions`
- All the "@blabla" are added in the `mentions` array, but they are displayed as a link only if they reffer to an existing user (bech32 address or existing NS token name)
- On a user profile, if you create a post, the user will be mentioned at the end of the post, only it's not your profile.
- On a user profile, we can see also the posts created by this user

### Hashtags
- We can add a hashtag by writting "#blabla"
- Then, on https://deploy-preview-318--testitori.netlify.app/feed/hashtag/blabla, we'll see the posts that contain `"#blabla"` in their metadata `hashtags`

### Urgent issues to fix
- Article creation: The arrays `mentions` and `hashtags` contains `""`, they are not empty even if the Article has mention or hashtag.
- Article creation : The hyperlink button doesn't work after the Article creation. => It doesn't add a `<a href`
- Some audios and videos come with an empty `url`, but they work when Post creation.
- We need to check that the files are correctly uploaded (Using the IPFS key getter and upload on Pinata)

### Less urgent, but TODO
- The navigation (The back buttons on screens) doesn't work well. It's a small point, but can increase well the experience.
- While loading a post on https://deploy-preview-318--testitori.netlify.app/feed/post/{postId}, we can see "Post not found". => Use a `loading` flag.
- Code Factorization
- A better responsive
- Animations
- Article creation : The scroll must follow the redaction. Actually, the scroll is always on top, so we need to scroll manually when writting.
- The `NewFeedInput` doesn't shrink when deleting rows while writting a Post on http://deploy-preview-318--testitori.netlify.app/feed
- Some UI adjustments

### Welcome upgrades
- A tip count => I think we'll send an `amount` and a `post.id` to the backend using `useTeritoriSocialFeedReactPostMutation`
- More tools for Article creation (There are unused existing tools in `draft-js`)
- Render preview for any URL (Like on Medium) (See the OpenGraph point above)
- A better Rich Text editor with a better `InlineToolbar`
- A draft system to retrieve unpublished content
- Test NTFStorage UCAN (We want that instead of Pinata)

### CODE REVIEW

**Review by YoLoADR — DISMISSED** [review]

File: packages/components/NewsFeed/CommentInput.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                height: 32,
+                width: 32,
+                borderRadius: 24,
+                backgroundColor: neutral33,
+                alignItems: "center",
+                justifyContent: "center",
+                marginRight: 20,
+              }}
+            >
+              <SVG source={cameraSVG} width={15} />
```
[the remark this answers is not in the document]
Comment (n0izn0iz) [reply packages/components/NewsFeed/CommentInput.tsx]: actually having the space in the svgs makes it harder to use

File: declarations.d.ts [hunk 2]
```diff
@@ -10,3 +10,5 @@ declare module "*.png" {
 
   export = value;
 }
+
+declare module "nft.storage/dist/bundle.esm.min.js";
```
Comment (n0izn0iz) [inline declarations.d.ts:14]: why do you need this?

File: packages/client-creators/socialFeedClient.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  contractAddress?: string;
+}
+
+const cachedClients: { [key: string]: TeritoriSocialFeedClient } = {};
+
+export const socialFeedClient = async ({
+  walletAddress,
+  contractAddress = process.env.TERITORI_SOCIAL_FEED_CONTRACT_ADDRESS || "",
+}: SocialFeedClientParams) => {
+  const socialFeedContractAddress = contractAddress.startsWith("tori-")
```
Comment (n0izn0iz) [inline packages/client-creators/socialFeedClient.ts:18]: `tori-` prefixed strings are ids not addresses

File: declarations.d.ts (hunk 2 above)
Comment (sakul-budhathoki) [reply declarations.d.ts:14]: Webpack has some issues while importing nft.storage
https://github.com/nftstorage/nft.storage/issues/876#issuecomment-985687135

`import { NFTStorage, File } from "nft.storage/dist/bundle.esm.min.js";
`
For this import we have to declare it on declaration file.

File: packages/components/NewsFeed/CommentInput.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        mainContainerStyle={{
+          paddingVertical: 12,
+          paddingHorizontal: 20,
+          backgroundColor: neutral22,
+          flexDirection: "row",
+        }}
+      >
+        <SVG height={12} width={12} source={penSVG} />
+
+        <TextInput
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated

File: packages/components/NewsFeed/CommentInput.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  return (
+    <View style={style}>
+      <TertiaryBox
+        fullWidth
+        height={64}
+        style={{
+          zIndex: 9,
+        }}
+        mainContainerStyle={{
+          paddingVertical: 12,
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/CommentInput.tsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+          paddingHorizontal: 20,
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/CommentInput.tsx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+          multiline
+          style={[
+            fontSemibold16,
+            {
+              width: "100%",
+              color: "white",
+              marginLeft: 12,
+
+              //@ts-ignore
+              outlineStyle: "none",
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/CommentInput.tsx [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+          ]}
+        >
+          {inputText.length}/{WORD_MAX_LIMIT}
+        </BrandText>
+      </TertiaryBox>
+      {isPostFooter && (
+        <View
+          style={{
+            backgroundColor: neutral17,
+            paddingTop: 24,
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/CommentInput.tsx [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+            paddingBottom: 12,
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/CommentInput.tsx [hunk 10]
```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from "react";
+import { TextInput, View, ViewStyle, TouchableOpacity } from "react-native";
+
+import cameraSVG from "../../../assets/icons/camera.svg";
+import dollarSVG from "../../../assets/icons/dollar-neutral77.svg";
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/CommentInput.tsx [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+            borderRadius: 8,
+          }}
+        >
+          <View
+            style={{
+              flexDirection: "row",
+              alignItems: "center",
+            }}
+          >
+            <SVG source={dollarSVG} height={17} />
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/CommentInput.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/CommentInput.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/PostInputBox.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+          backgroundColor: neutral11,
+          flexDirection: "row",
+          alignItems: "center",
+          borderBottomWidth: 1,
+          borderColor: neutral33,
+        },
+        style,
+      ]}
+    >
+      <TextInput
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/PostInputBox.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/NewsFeed/PostInputBox.tsx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        placeholder="Hey yo! Post something here!___"
+        placeholderTextColor={neutral77}
+        style={[
+          fontSemibold16,
+          {
+            width: "100%",
+            color: "white",
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/NewsFeed/PostInputBox.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/WalletStatusBox.tsx [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+import { SuccessBadge } from "./badges/SuccessBadge";
+import { TertiaryBox } from "./boxes/TertiaryBox";
+
+export const WalletStatusBox: React.FC = () => {
+  const selectedWallet = useSelectedWallet();
+
+  return (
+    <TertiaryBox
+      fullWidth
+      style={{}}
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/WalletStatusBox.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/cards/CommentsContainer.tsx [hunk 15]
```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { StyleProp, View, ViewStyle } from "react-native";
+
+import { neutral22 } from "../../utils/style/colors";
+import { SocialCommentCard } from "./SocialCommentCard";
+
+export const CommentContainer: React.FC<{
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/cards/CommentsContainer.tsx]: Outdated, this file doesn't exist anymore

File: packages/components/cards/CommentsContainer.tsx [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+        }}
+      />
+      <SocialCommentCard comment={comment} />
+    </View>
+  );
+};
+
+export const CommentsContainer: React.FC<{
+  comments: object[];
+  style?: StyleProp<ViewStyle>;
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/components/cards/CommentsContainer.tsx]: Outdated, this file doesn't exist anymore

File: packages/screens/OrganizationPublicProfile/OrganizationPublicProfileScreen.tsx [hunk 17]
```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from "react";
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/screens/OrganizationPublicProfile/OrganizationPublicProfileScreen.tsx]: Removed

File: packages/hooks/useTNSMetadata.ts [hunk 18]
```diff
@@ -1,40 +1,60 @@
...
-      });
-
-      // ======== Getting NFT info
-      const nftInfo = await tnsClient.nftInfo({
-        tokenId: aliasResponse.username,
-      });
-
-      return { tokenId: aliasResponse.username, ...nftInfo.extension };
+      const storedMetadata = metaDataList[address];
+      if (storedMetadata) return storedMetadata;
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/hooks/useTNSMetadata.ts]: Outaded, this file doesn't exist anymore

**Comment by omniwired** [comment]
Please make this blue line on top of gray one.
image

image

**Comment by omniwired** [comment]
if possible let's check if the videos are not downloaded as they appear, but only when the user click play on them.
It would be nice to add a thumbnail as well

File: packages/utils/style/layout.ts [hunk 19]
```diff
@@ -10,6 +10,18 @@ export const walletSelectorWidth = 220;
 export const smallSidebarWidth = 76;
 export const fullSidebarWidth = 210;
 
+export const getResponsiveScreenContainerMarginHorizontal = (width: number) => {
```
[the remark this answers is not in the document]
Comment (WaDadidou) [reply packages/utils/style/layout.ts:17]: We need to check the usage of `getResponsiveScreenContainerMarginHorizontal`. I'm not sure about it.
We need reponsive for Social Feed

File: declarations.d.ts (hunk 2 above)
Comment (WaDadidou) [reply declarations.d.ts:14]: `packages/candymachine/nft-storage-upload.ts` removed since we're using `packages/candymachine/pinata-upload.ts`

**Review by zxxma — APPROVED** [review]
LGTM 👍

**Pull request merged**